src/batch-builder.js

```diff
@@ -567,148 +563,68 @@ module.exports = class BatchBuilder {
             ]);
         }
 
-        if (op2=="INSERT") {
-            // State 2
-            this.input.sign2[i] = 0x1234;    // It should not matter
-            this.input.ay2[i] = 0x1234;      // It should not matter
-            this.input.balance2[i] = 0x1234; // It should not matter
-            this.input.nonce2[i] = 0x1234;   // It should not matter
-            this.input.tokenID2[i] = 0x1234; // It should not matter
-            this.input.ethAddr2[i] = 0x1234; // It should not matter
-
-            // if original amount is 0, leaf is not inserted in the exit tree
-            if (!Scalar.eq(tx.amount, 0)){
-                const newValue = stateUtils.hashState(newState2);
-
-                const res = await this.exitTree.insert(tx.fromIdx, newValue);
-                if (res.found) {
-                    throw new Error("Invalid Exit account");
-                }
-                let siblings = res.siblings;
-                while (siblings.length<this.nLevels+1) siblings.push(Scalar.e(0));
+        let updateToIdx;
+        if (isExit){
+            updateToIdx = tx.fromIdx;
+        } else {
+            updateToIdx = finalToIdx;
+        }
 
-                this.input.newExit[i] = 1;   // must be 1 to signal new exit leaf
-                this.input.siblings2[i] = siblings;
-                this.input.isOld0_2[i] = res.isOld0 ? 1 : 0;
-                this.input.oldKey2[i] = res.isOld0 ? 0 : res.oldKey;
-                this.input.oldValue2[i] = res.isOld0 ? 0 : res.oldValue;
+        if (op2 == "UPDATE") {
```

Comment:
Exit operation could be performed in OP1 since are changes in the same leaf. Could this be more efficient?

Reply:
Yes, it could be done in such a way. However, this it does not provide more efficiency in terms of constraints since `processor2` should be added in all transactions in the circuit in order to perform standard transfers. Therefore, the number of constraints would be the same.
Said that, both approaches are totally valid in exits, either update both balances in `processor1` or just update `balance` in `processor1` and `exitBalance` in `processor2`. I took the second approach since it is more similar to the one that we have previously.